File: longest_palindrome.js
function longestPalindrome(s) {
    let maxLen = 0;
    for (let i = 0; i < s.length; i++) {
        for (let j = i; j < s.length; j++) {
            let cur = s.substring(i, j + 1);
            if (cur.split('').reverse().join('') === cur){
              maxLen = Math.max(maxLen, cur.length);
            }
        }
    }
    return maxLen;
}
let s = "racecar"
console.log(longestPalindrome(s));

File: longest_palindrome.py
class Solution(object):

    def longestPalindrome(self, s):
        strLen = len(s)
        maxLen = 0
        allSubStrings = []
        for i in xrange(strLen):
            for j in xrange(i, strLen):
                allSubStrings.append(s[i: j + 1])

        for k in allSubStrings:
            if k[::-1] == k:
                maxLen = max(maxLen, len(k))
        return maxLen

# s = "string"
s = "racecar"
print Solution().longestPalindrome(s)
